### ISSUE TITLE
Interop functions does not return error values

### ISSUE THREAD

**Issue by ThisaruGuruge** [ballerina-platform/ballerina-lang#19893]
**Description:**
When we return an Error from an inter-op function, it isn't passing to ballerina function. 

**Steps to reproduce:**

Following function should print the error but it prints `No Error` instead. (The example code may contain syntax errors, ignore them).

Ballerina function:
```ballerina
public type Listener Object {
    public function __init(TimerConfiguration|AppointmentConfiguration configs) {
        if (configs is TimerConfiguration) {
            if (configs["initialDelayInMillis"] == ()) {
                configs.initialDelayInMillis = configs.intervalInMillis;
            }
        }
        self.listenerConfiguration = configs;
        var result = initExternal(self);
        if (result is ListenerError) {
            panic result;
        }
    }
}

function initExternal(Listener task) returns ListenerError? = @java:Method {
    name: "init",
    class: "org.ballerinalang.stdlib.task.actions.TaskActions"
} external;
```

Java:

```java
import static org.ballerinalang.jvm.util.BLangConstants.BALLERINA_BUILTIN_PKG_PREFIX;

public class TaskActions {

    public static final String PACKAGE_NAME = "task";
    public static final String LISTENER_ERROR = "{ballerina/task}ListenerError";
    public static final BPackage TASK_PACKAGE_ID = new BPackage(BALLERINA_BUILTIN_PKG_PREFIX, PACKAGE_NAME);
    public static final String DETAIL_RECORD_NAME = "Detail";

    public static Object init(ObjectValue taskListener) {
        try {
            throw new SchedulingException("Scheduling Exception");
        } catch (SchedulingException e) {
           return BallerinaErrors.createError(e.getMessage());
        }
    }

    public static ErrorValue createTaskError(String message) {
        MapValue<String, Object> detail = createTaskDetailRecord(message);
        return BallerinaErrors.createError(LISTENER_ERROR, detail);
    }

    private static MapValue<String, Object> createTaskDetailRecord(String message) {
        MapValue<String, Object> detail = BallerinaValues.createRecordValue(TASK_PACKAGE_ID, DETAIL_RECORD_NAME);
        return BallerinaValues.createRecord(detail, message, null);
    }
```

**Affected Versions:**
> Branch: Ballerina-1.0.x

**Comment by SupunS** [comment]
This is working in the master. Is it possible to use master branch?

**Comment by ThisaruGuruge** [comment]
Okay, will change the branch.

**Comment by ThisaruGuruge** [comment]
This is not fixed in the master too. The error message is invalid.

**Comment by ThisaruGuruge** [comment]
This is fixed now.

**Comment by SupunS** [comment]
Fixed with #19703
